src/Configuration.h

```diff
@@ -46,10 +46,22 @@
 #  endif
 
 // Enables the ThingsBoard class to be fully dynamic instead of requiring template arguments to statically allocate memory.
-// If enabled the programm might be slightly slower and all the memory will be placed onto the heap instead of the stack.
+// If enabled the program might be slightly slower and all the memory will be placed onto the heap instead of the stack.
 // See https://arduinojson.org/v6/api/dynamicjsondocument/ for the main difference in the underlying code.
 # ifndef THINGSBOARD_ENABLE_DYNAMIC
 #   define THINGSBOARD_ENABLE_DYNAMIC 0
 # endif
 
+// Enables the ThingsBoard class to save the allocated memory of the DynamicJsonDocument into psram instead of onto the sram.
+// Enabled by default if THINGSBOARD_ENABLE_DYNAMIC has been set, because it requries DynamicJsonDocument to work.
+// If enabled the program might be slightly slower and all the memory will be placed onto psram instead of sram.
+// See https://arduinojson.org/v6/api/basicjsondocument/ for the main difference in the underlying code.
+# if THINGSBOARD_ENABLE_DYNAMIC
+#   ifndef THINGSBOARD_ENABLE_PSRAM
```

Comment:
Probably we also should check for __has_include(<esp_heap_caps.h>), because we can enable dynamic on boards without PSRAM, what do you think?

Reply:
Sounds like a good idea will fix that